libraries/WiFi/src/WiFiSTA.cpp

```diff
@@ -391,6 +391,25 @@ bool WiFiSTAClass::disconnect(bool wifioff, bool eraseap)
     return false;
 }
 
+/**
+ * Disconnect from the network.
+ * @param wifioff `true` to turn the Wi-Fi radio off.
+ * @param eraseap `true` to erase the AP configuration from the NVS memory.
+ * @param timeoutLength timeout to wait for status change
+ * @return `true` when successful.
+ */
+bool WiFiSTAClass::disconnect(bool wifioff, bool eraseap, unsigned long timeoutLength)
+{
+  bool rv = disconnectAsync(wifioff, eraseap);
+  if (!rv)
+      return false;
+  unsigned long start = millis();
+  while (status() == WL_CONNECTED && (millis() - start) < timeoutLength) {
+      delay(100);
```

Comment:
Shouldn't this `100` be something like `timeoutLength < 100 ? timeoutLength : 100` ?

Reply:
it is copied from `waitForConnectResult`

Reply:
```cpp
uint32_t delay_time = timeoutLength < 10 ? timeoutLength : 10;
while ((WiFiGenericClass::getStatusBits() & STA_CONNECTED_BIT) != 0 && (millis() - start) < timeoutLength) {
  delay(delay_time);
}
```